### ISSUE TITLE
not working anymore

### ISSUE THREAD

**Issue by thorizer** [master131/BlockTheSpot#6]
not working with version 1.0.96.181
installed it manually on windows 10

**Comment by themikeambrose** [comment]
I just installed using the automatic (.bat) method and it works fine. Latest version also.

**Comment by thorizer** [comment]
i tried the automatic too and it didn't work. ads are everywhere

**Comment by scorthyn** [comment]
1.0.96.181 is working fine people, tried both on my laptop and desktop, something to do with your system \ instalation. If you used other adblockers like ez adblock then it messes with this one.

**Comment by KaleidonKep99** [comment]
Install it manually by dropping the file in **%appdata%\Spotify**.
I don't really recommend using the automatic installation via the batch file, since it seems to be unreliable.

**Comment by master131** [comment]
@KaleidonKep99 I've updated the automatic installation method to detect the Spotify installation path via the registry. Hopefully that makes it a bit more reliable.

@thorizer Is Spotify crashing or is the adblocking simply not working? If the adblock not working, can you please download ListDLLs with Spotify running in the background and post the output of this command?
`listdlls Spotify.exe`

https://docs.microsoft.com/en-us/sysinternals/downloads/listdlls

**Comment by SciallaMan** [comment]
> @KaleidonKep99 I've updated the automatic installation method to detect the Spotify installation path via the registry. Hopefully that makes it a bit more reliable.
> 
> @thorizer Is Spotify crashing or is the adblocking simply not working? If the adblock not working, can you please download ListDLLs with Spotify running in the background and post the output of this command?
> `listdlls Spotify.exe`
> 
> https://docs.microsoft.com/en-us/sysinternals/downloads/listdlls

I've the same problem (doesn't block ads) so I did what you said and this is the log. Thanks in advance.
[ListDlls Log.txt](https://github.com/master131/BlockTheSpot/files/2732448/ListDlls.Log.txt)

**Comment by StefanAndreea** [comment]
I just installed manually and it works. Version 1.0.96.181.gf6bc1b6b

**Comment by SciallaMan** [comment]
> I just installed manually and it works. Version 1.0.96.181.gf6bc1b6b

What do you see in your Appdata spotify folder? I see it empty, in the guide it says that there should be the spotify.exe. Are you on win 10 too?

Edit: Seems to be a problem related to installation path, in automatic installation it says 
`Spotify installation was not detected, falling back to default path. Installation complete.`

**Comment by StefanAndreea** [comment]
> 
> 
> > I just installed manually and it works. Version 1.0.96.181.gf6bc1b6b
> 
> What do you see in your Appdata spotify folder? I see it empty, in the guide it says that there should be the spotify.exe. Are you on win 10 too?
> 
> Edit: Seems to be a problem related to installation path, in automatic installation it says
> `Spotify installation was not detected, falling back to default path. Installation complete.`

Yes I'm on win 10. The _%APPDATA%/Spotify_ folder looks just like a common installation folder, with subfolders, other files and spotify.exe . Maybe you should reinstall Spotify and make sure you deleted the user files and everything.

**Comment by SciallaMan** [comment]
> > > I just installed manually and it works. Version 1.0.96.181.gf6bc1b6b
> > 
> > 
> > What do you see in your Appdata spotify folder? I see it empty, in the guide it says that there should be the spotify.exe. Are you on win 10 too?
> > Edit: Seems to be a problem related to installation path, in automatic installation it says
> > `Spotify installation was not detected, falling back to default path. Installation complete.`
> 
> Yes I'm on win 10. The _%APPDATA%/Spotify_ folder looks just like a common installation folder, with subfolders, other files and spotify.exe . Maybe you should reinstall Spotify and make sure you deleted the user files and everything.

I'll try, did you downloaded it from store or website? It seems to be the same version but maybe...

**Comment by KaleidonKep99** [comment]
You should add a warning in the readme file about this mod not supporting the Windows Store release of Spotify.

**Comment by master131** [comment]
I was not aware a Windows Store version existed. Please attempt to use the standard desktop release and see if it works.

**Comment by SciallaMan** [comment]
Tried on another Pc and it works, the problem is related to the Store's version. Sadly on the first Pc even if I download it from the website it installs the store version automatically.

**Comment by master131** [comment]
@SciallaMan Uninstall the Microsoft Store version (search in the Start Menu and right-click > Uninstall) and try installing the Desktop version from here:

https://download.scdn.co/SpotifyFullSetup.exe

**Comment by SciallaMan** [comment]
> @SciallaMan Uninstall the Microsoft Store version (search in the Start Menu and right-click > Uninstall) and try installing the Desktop version from here:
> 
> https://download.scdn.co/SpotifyFullSetup.exe

Tomorrow I'll try on the laptop, thank you in advance!

Edit: @master131 I tried what you said but nothing. It is very strange, when I install it the icon is the Desktop version's icon, then it says "Data migration..." with a progress bar and it automatically becomes the Store version, the icon change and in the AppData there is nothing. I tried also to do that offline but it is the same...the appData folder populate for some seconds then it migrate al data to store's version in another folder (I found it but the structure is very different, there isn't the spotify.exe).

**Comment by SciallaMan** [comment]
> > @SciallaMan Uninstall the Microsoft Store version (search in the Start Menu and right-click > Uninstall) and try installing the Desktop version from here:
> > https://download.scdn.co/SpotifyFullSetup.exe
> 
> Tomorrow I'll try on the laptop, thank you in advance!

Edit: @master131 I tried what you said but nothing. It is very strange, when I install it the icon is the Desktop version's icon, then it says "Data migration..." with a progress bar and it automatically becomes the Store version, the icon change and in the AppData there is nothing. I tried also to do that offline but it is the same...the appData folder populate for some seconds then it migrate al data to store's version in another folder (I found it but the structure is very different, there isn't the spotify.exe).

**Comment by master131** [comment]
@SciallaMan I'm not sure, sorry. There are a few people reporting similar issues online but none have a solution.

https://community.spotify.com/t5/Desktop-Windows/Is-there-a-non-windows-store-version/td-p/4563408
https://old.reddit.com/r/Windows10/comments/9kojxv/spotify_installation_not_from_store/

EDIT: @SciallaMan Give this a try:
https://community.spotify.com/t5/Desktop-Windows/How-to-get-the-original-spotify-back-not-migrating-the-data-to/td-p/4500480

**Comment by SciallaMan** [comment]
> @SciallaMan I'm not sure, sorry. There are a few people reporting similar issues online but none have a solution.
> 
> https://community.spotify.com/t5/Desktop-Windows/Is-there-a-non-windows-store-version/td-p/4563408
> https://old.reddit.com/r/Windows10/comments/9kojxv/spotify_installation_not_from_store/
> 
> EDIT: @SciallaMan Give this a try:
> https://community.spotify.com/t5/Desktop-Windows/How-to-get-the-original-spotify-back-not-migrating-the-data-to/td-p/4500480

It was enough to do the first three steps of the guide and the original version came back. Thank you very much! The problem was that I didn't remember to have a second account in which the store's version was still installed.  Now "netutils" do their work properly!

**Comment by master131** [comment]
Due to a lack of response from the OP, I'm closing the issue.

**Comment by thorizer** [comment]
sorry for the late feedback. I'm not using the store version. it worked for one week then stopped forever. the file is still there in spotify directory. this is the log from listdll.exe 
[spotify dll.txt](https://github.com/master131/BlockTheSpot/files/2814826/spotify.dll.txt)
maybe because i'm not from eu//na and spotify uses other domains ?

**Comment by master131** [comment]
The hosts blocking is just a backup feature and isn't the main method to block ads. Can you try fully uninstalling Spotify (so cache files and stuff are removed) and try re-installing it again? Also make sure you're using the latest BlockTheSpot build as well.

**Comment by Ede123** [comment]
JFTR (sorry for being slightly off topic wrt to the original issue, but I hope it saves others some time):

I've also had the problem of Spotify
1. automatically installing the Store version instead of the normal desktop version
2. even when updating from an older non-Store version eventually removing the desktop version during an update and migrating to the Store version instead.

I've found both issues can be worked around:
1. is solved by running the installer in compatibility mode for Windows 7
2. is solved by running the following command in a powershell window launched with administrative privileges and restarting the system afterwards:  
  `Get-AppxPackage -AllUsers *spotify* | Remove-AppxPackage  -AllUsers`  
  (this is basically a condensed and more systematic version of what was linked above. I'm not sure if a restart is actually required, but I did it in my case just to be sure; removing the Store version via control panel and without restart did *not* work)

**Comment by OzakIOne** [comment]
@Ede123 Also having Windows 10 Enterprise LTSC, might fix the problem of store version because the microsoft store is removed by default on this edition so i don't think spotify could migrate your installation from desktop to store version

**Comment by jnslsrwy** [comment]
I tried modifying GPOs regarding the Store without success.
In my case the installation in `%appdata%` would be deleted as soon as as the installer finishes as Spotify would migrate directly to the Store-Edition. Therefore I did the following:
1. boot into safe mode
2.  install Spotify with latest installer from [Spotify CDN](https://download.scdn.co/SpotifySetup.exe)
3. remove `SpotifyMigrator.exe` & `SpotifyStartupTask.exe` from `%appdata%\Spotify` after setup completes
4. reboot
5. as @Ede123 suggested:
   > running the following command in a powershell window launched with administrative privileges and restarting the system afterwards:
    `Get-AppxPackage -AllUsers *spotify* | Remove-AppxPackage  -AllUsers`
6. install @master131 's BlockTheSpot
